crates/test/migration-test/tests/network_integration.rs

```diff
@@ -0,0 +1,403 @@
+fn load_wallet_into_tmpdir() -> TempDir {
+    let tmpdir = tempdir().unwrap();
+    let mut setup_cmd = get_pcli_path();
+    setup_cmd
+        .args([
+            "--home",
+            tmpdir.path().to_str().unwrap(),
+            "init",
+            "--grpc-url",
+            &PD_NODE_URL,
+            "soft-kms",
+            "import-phrase",
+        ])
+        .write_stdin(SEED_PHRASE)
+        .timeout(std::time::Duration::from_secs(TIMEOUT_COMMAND_SECONDS));
+    setup_cmd
+        .assert()
+        .stdout(predicate::str::contains("Writing generated config"));
+    tmpdir
+}
+
+/// Sync the wallet. Fallible, so we can use it so confirm network is halted.
+fn sync(tmpdir: &TempDir) -> anyhow::Result<()> {
+    let mut sync_cmd = get_pcli_path();
+    sync_cmd
+        .args(["--home", tmpdir.path().to_str().unwrap(), "view", "sync"])
+        .timeout(std::time::Duration::from_secs(TIMEOUT_COMMAND_SECONDS));
+    sync_cmd.assert().try_success()?;
+    Ok(())
+}
+
+/// Look up a currently active validator on the testnet.
+/// Will return the most bonded, which means the Penumbra Labs CI validator.
+fn get_validator(tmpdir: &TempDir) -> String {
+    let mut validator_cmd = get_pcli_path();
+    validator_cmd
+        .args([
+            "--home",
+            tmpdir.path().to_str().unwrap(),
+            "query",
+            "validator",
+            "list",
+        ])
+        .timeout(std::time::Duration::from_secs(TIMEOUT_COMMAND_SECONDS));
+    validator_cmd.assert().success();
+
+    // Pull out one of the validators from stdout.
+    let stdout_vec = validator_cmd.unwrap().stdout;
+    let validator_regex = Regex::new(r"penumbravalid1\w{58}").unwrap();
+    let captures = validator_regex.captures(std::str::from_utf8(&stdout_vec).unwrap());
+
+    // We retrieve the first match via index 0, which results in most trusted.
+    captures.unwrap()[0].to_string()
+}
+
+/// Look up current block height on target network.
+fn get_current_height(tmpdir: &TempDir) -> u64 {
+    let mut info_cmd = get_pcli_path();
+    info_cmd
+        .args([
+            "--home",
+            tmpdir.path().to_str().unwrap(),
+            "query",
+            "chain",
+            "info",
+        ])
+        .timeout(std::time::Duration::from_secs(TIMEOUT_COMMAND_SECONDS));
+    info_cmd.assert().success();
+
+    let stdout_vec = info_cmd.unwrap().stdout;
+    let height_regex = Regex::new(r"Current Block Height\s+(\d+)").unwrap();
+    let captures = height_regex.captures(std::str::from_utf8(&stdout_vec).unwrap());
+
+    // Reminder: index 0 is entire match, index 1 is first captured group.
+    captures.unwrap()[1]
+        .parse()
+        .expect("block height in pcli output is u64")
+}
+
+// Needed functions:
+//
+//   get_current_height()
+//   generate_proposal
+//   submit_proposal
+//
+//
+
+/// Look up top validator and delegate `TEST_DELEGATION` to it.
+#[cfg_attr(not(feature = "migration-test"), ignore)]
```

Comment:
the line as is, says "_conditionally apply the `#[ignore]` attribute if the `migration-test` feature is **not** enabled._"

we can instead say "_this is active if the `migration-test` feature is enabled_"

```suggestion
#[cfg(feature = "migration-test")]
```